xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/ClassLoaderSkin.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xpn.xwiki.internal.skin;
+
+import java.net.URL;
+
+import javax.inject.Provider;
+
+import org.slf4j.Logger;
+import org.xwiki.url.URLConfiguration;
+
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Represents a skin than can be access using the classloader (for instance, stored in a jar).
+ *
+ * @version $Id$
+ * @since 13.8RC1
+ * @since 13.7.1
+ * @since 13.4.4
+ */
+public class ClassLoaderSkin extends AbstractResourceSkin
+{
+    private final ClassLoader classloader;
+
+    private final Provider<XWikiContext> xcontextProvider;
+
+    private final URLConfiguration urlConfiguration;
+
+    /**
+     * Default constructor.
+     *
+     * @param id the skin id (for instance, {@code "flamingo"})
+     * @param skinManager the skin manager that instantiates this skin
+     * @param configuration the skin internal configuration, used to access the default parent skin id
+     * @param logger a logger used to log warning in case of error when parsing a skin's syntax
+     * @param xcontextProvider a wiki context provide, used to give access to the context when resolving the skin's
+     *     rsources.
+     * @param urlConfiguration the url configuration used to resolve the url of the skin's resources
+     */
+    public ClassLoaderSkin(String id, InternalSkinManager skinManager, InternalSkinConfiguration configuration,
+        Logger logger, Provider<XWikiContext> xcontextProvider,
+        URLConfiguration urlConfiguration)
+    {
+        super(id, skinManager, configuration, logger);
+
+        this.classloader = Thread.currentThread().getContextClassLoader();
```

Comment:
Since the ClassLoaderSkin is kept in a cache, using whatever is the current thread at the moment of the ClassLoaderSkin instance creation is not very safe. Would be safer to dynamically ask ClassLoaderManager for the right ClassLoader for the current wiki instead in `getResourceURL`.

Reply:
Should I use a particular `namespace` when calling `org.xwiki.classloader.ClassLoaderManager#getURLClassLoader` or is `null` fine?

Reply:
As I said, you should ask the classloader for the current wiki, so something like `new WikiNamespace(this.wikiDescriptorManager.getCurrentWikiId()))`.